### PR TITLE
Don't override user preferences in local storage

### PR DESCRIPTION
**Pull request by abeddow91**
## What does this change?
Removes comment context state that is not needed and caused a bug where user preferences were being overriden.

## Why?
Previously, directing to a page with a discussion comment hash triggered an api call with user filters provided as a param which then returned filters. These filters were then stored as the user defined filters. Best case scenario we are restoring state which is already stored as the user preferences weren't changed. Worse case scenario we are overriding user preferences without being asked to do so.

Now, only changes to the filters made by a user are saved to disk. Any overrides are applied in a mapping function locally before filters are passed down from discussions.

This also fixes a bug where the ordering filter was being overriden by the wrong orderby type. This is because the API uses  and returns "mostRecommended" instead of "recommendations"

### CODE REVIEW

File: dotcom-rendering/src/components/Discussion.tsx [hunk 1]
```diff
@@ -126,16 +122,13 @@ export const Discussion = ({
 	}, [filters, commentPage, shortUrlId]);
 
 	useEffect(() => {
-		const orderByClosed = isClosedForComments ? 'oldest' : undefined;
```
Comment (abeddow91) [inline dotcom-rendering/src/components/Discussion.tsx:129]: I've removed this as this is not working as expected. Ive created a ticket [here](https://github.com/orgs/guardian/projects/88?pane=issue&itemId=51173252) to look into this.

File: dotcom-rendering/src/components/Discussion.tsx [hunk 2]
```diff
@@ -186,19 +172,6 @@ export const Discussion = ({
...
-		}
-	}, []);
-
-	useEffect(() => {
-		// There's no point showing the view more button if there isn't much more to view
-		if (commentCount === 0 || commentCount === 1 || commentCount === 2) {
-			setIsExpanded(true);
-		}
-	}, [commentCount]);
-
```
Comment (abeddow91) [inline dotcom-rendering/src/components/Discussion.tsx:201]: These lines are duplicated above so I'm removing.

File: dotcom-rendering/src/components/Discussion.tsx [hunk 3]
```diff
@@ -229,8 +211,11 @@ export const Discussion = ({
 					idApiUrl={idApiUrl}
 					page={commentPage}
 					setPage={setCommentPage}
-					filters={filters}
-					setFilters={setFilters}
+					filters={validFilters}
+					setFilters={(newFilters) => {
+						setHashCommentId(undefined);
```
Comment (mxdvl) [inline dotcom-rendering/src/components/Discussion.tsx:216]: When they set filters, the user is no longer looking at a specific comment, so we can reset the `hashCommentId` value.

File: dotcom-rendering/src/components/Discussion.tsx [hunk 4]
```diff
@@ -71,16 +70,19 @@ const commentIdFromUrl = () => {
 	return parseInt(commentId, 10);
 };
 
-const filterByPermalinks = (
-	threads: ThreadsType,
+const remapFilters = (
```
Comment (mxdvl) [inline dotcom-rendering/src/components/Discussion.tsx:73]:
```suggestion
const remapFilersToValidCombination = (
```

**Review by mxdvl — APPROVED** [review]

**Pull request merged**